### PR TITLE
rebase: retain old and new constants, .github folder changes, fix errors (1/?)

### PR DESCRIPTION
**Pull request by jaideepr97**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind chore
> /kind cleanup
> /kind failing-test
> /kind enhancement
> /kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:
1/(?) PRs to rebase the redesgin branch

This PR does:
- rewrite the common packages to organize constants in a better way
- add the `.github` folder changes
- undo some changes made to existing code and fix errors
- moves `argoutil` to `pkg` folder

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: common/envVars.go [hunk 1]
```diff
@@ -43,4 +43,7 @@ const (
 	// ArgoCDClusterConfigNamespacesEnvVar is the environment variable that contains the list of namespaces allowed to host cluster config
 	// instances
 	ArgoCDClusterConfigNamespacesEnvVar = "ARGOCD_CLUSTER_CONFIG_NAMESPACES"
+
+	// ArgoCDLabelSelectorEnvVar is an environment variable that contains the labels used for selective instance reconilliation.
+	ArgoCDLabelSelectorEnvVar = "ARGOCD_LABEL_SELECTOR"
```
Comment (ciiay) [inline common/envVars.go:48]: Just to confirm, when common/TOBEREMOVED.go gets removed, `ArgoCDLabelSelectorKey` will be replaced by `ArgoCDLabelSelectorEnvVar` right? There would be another rebase PR to update all the constants to use the new name?

**Review by ciiay — APPROVED** [review]
LGTM
Great job 👏 
Just left a small comment.

File: .github/workflows/ci-build.yaml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.19
+          go-version: '1.20'
```
Comment (ciiay) [inline .github/workflows/ci-build.yaml:35]: Do we need to upgrade go version to 1.20 in go.mod to match these updates?

File: common/envVars.go (hunk 1 above)
Comment (jaideepr97) [reply common/envVars.go:48]: yes, we'll handle those cases later

File: .github/workflows/ci-build.yaml (hunk 2 above)
Comment (jaideepr97) [reply .github/workflows/ci-build.yaml:35]: its upgraded in a later PR

**Pull request merged**